tests/unit/python/foglamp/common/test_statistics.py

```diff
@@ -24,92 +23,108 @@
 @pytest.allure.story("common", "statistics")
 class TestStatistics:
 
-    def test_init_with_no_storage(self):
+    async def test_init_with_no_storage(self):
         storage_client_mock = None
         with pytest.raises(TypeError) as excinfo:
-            Statistics(storage_client_mock)
-        assert str(excinfo.value) == 'Must be a valid Storage object'
+            await statistics.create_statistics(storage_client_mock)
+        assert str(excinfo.value) == 'Must be a valid Async Storage object'
 
-    def test_init_with_storage(self):
+    async def test_init_with_storage(self):
         storage_client_mock = MagicMock(spec=StorageClient)
-        s = Statistics(storage_client_mock)
-        assert isinstance(s, Statistics)
+        s = await statistics.create_statistics(storage_client_mock)
+        assert isinstance(s, statistics.Statistics)
         assert isinstance(s._storage, StorageClient)
 
-    def test_singleton(self):
+    async def test_singleton(self):
         """ Test that two statistics instance share the same state """
         storageMock1 = MagicMock(spec=StorageClient)
-        s1 = Statistics(storageMock1)
+        s1 = await statistics.create_statistics(storageMock1)
         storageMock2 = MagicMock(spec=StorageClient)
-        s2 = Statistics(storageMock2)
+        s2 = await statistics.create_statistics(storageMock2)
         assert s1._storage == s2._storage
 
-    def test_register(self):
+    async def test_register(self):
         """ Test that register results in a database insert """
         storageMock = MagicMock(spec=StorageClient)
-        stats = Statistics(storageMock)
-        loop = asyncio.get_event_loop()
-        loop.run_until_complete(stats.register('T1Stat', 'Test stat'))
-        args, kwargs = stats._storage.insert_into_tbl.call_args
-        assert args[0] == 'statistics'
-        expected_storage_args = json.loads(args[1])
-        assert expected_storage_args['key'] == 'T1Stat'
-        assert expected_storage_args['value'] == 0
-        assert expected_storage_args['previous_value'] == 0
-        assert expected_storage_args['description'] == 'Test stat'
+        stats = statistics.Statistics(storageMock)
+        stats._registered_keys = []
+        async def mock_coro():
+            return {"response": "updated", "rows_affected": 1}
+
+        with patch.object(stats._storage, 'insert_into_tbl', return_value=mock_coro()) as stat_update:
+            await stats.register('T1Stat', 'Test stat')
+            args, kwargs = stat_update.call_args
+            assert args[0] == 'statistics'
+            expected_storage_args = json.loads(args[1])
+            assert expected_storage_args['key'] == 'T1Stat'
+            assert expected_storage_args['value'] == 0
+            assert expected_storage_args['previous_value'] == 0
+            assert expected_storage_args['description'] == 'Test stat'
```

Comment:
Fix the indentation of stat_update assertions. Can't we use here 
`stat_update.assert_called_once_with("statistics", payload)` instead of individual args key-pair ?

Reply:
Tried but it fails randomly because of order of json received